conda/recipe/build.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+#
+# (C) Copyright IBM Corp. 2019. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Create build directory for cmake and enter it
+mkdir $SRC_DIR/build
+cd $SRC_DIR/build
+
+#Determine Architecture
+ARCH="powerpc64le-conda_cos7"
+ARCH_SHORTNAME="ppc64le"
+if [ `arch` = "x86_64" ]; then
+    ARCH="x86_64-conda_cos6"
+    ARCH_SHORTNAME="x86_64"
+fi
+
+# Create 'gcc' symlink so nvcc can find it
+ln -s $CONDA_PREFIX/bin/${ARCH}-linux-gnu-gcc $CONDA_PREFIX/bin/gcc
+
+# Add libjpeg-turbo location to front of CXXFLAGS so it is used instead of jpeg
+export CXXFLAGS="-I$CONDA_PREFIX/libjpeg-turbo/include ${CXXFLAGS}"
```

Comment:
We already have `-DJPEG_INCLUDE_DIR` in line 43. Is this export necessary?

Reply:
This export is here as a result of the whole jpeg vs jpeg-turbo scenario. Without this here, the build looks for the regular `jpeg` headers (`jpeg` conda package being brought into the build as a dependency of `opencv`) instead of the `jpeg-turbo` headers which are named the same. The compilation error resulting from this is as follows:
```
/root/anaconda3/conda-bld/dali_1565724018162/work/dali/image/jpeg_mem.cc: In function 'dali::uint8* dali::jpeg::{anon
ymous}::UncompressLow(const void*, dali::jpeg::{anonymous}::FewerArgsForCompiler*)':                                 
/root/anaconda3/conda-bld/dali_1565724018162/work/dali/image/jpeg_mem.cc:135:9: error: 'JCS_EXT_BGR' was not declared
 in this scope                                                                                                       
         JCS_EXT_BGR : JCS_RGB;                                                                                      
         ^~~~~~~~~~~                                                                                                 
/root/anaconda3/conda-bld/dali_1565724018162/work/dali/image/jpeg_mem.cc:135:9: note: suggested alternative: 'JCS_BG_
RGB'                                                                                                                 
         JCS_EXT_BGR : JCS_RGB;                                                                                      
         ^~~~~~~~~~~                                                                                                 
         JCS_BG_RGB                                                                                                  
/root/anaconda3/conda-bld/dali_1565724018162/work/dali/image/jpeg_mem.cc:490:31: warning: comparison of integer expre
ssions of different signedness: 'int' and 'JDIMENSION' {aka 'unsigned int'} [-Wsign-compare]                         
     if (argball->height_read_ > target_output_height) {                                                             
         ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~                                                                
make[2]: *** [dali/CMakeFiles/dali.dir/build.make:162: dali/CMakeFiles/dali.dir/image/jpeg_mem.cc.o] Error 1
```
As you can see above, `JCS_EXT_BGR` is unique to `jpeg-turbo`
Since `jpeg-turbo` is not a simple "swap and pop" of `jpeg` we could not just overwrite jpeg's headers instead opting to put jpeg-turbo in a unique location. Export is so build can find the new location.

Reply:
I solved this issue by providing a CMake prefix path instead (see comment below)